src/Microsoft.VisualStudio.Composition/ComposedPart.cs

```diff
@@ -160,6 +160,57 @@ public IEnumerable<ComposedPartDiagnostic> Validate(IReadOnlyDictionary<Type, Ex
             }
         }
 
+        /// <summary>
+        /// Checks whether this part contains an import with <see cref="ImportCardinality.ExactlyOne"/> cardinality that has been invalidated,
+        /// which would invalidate this part as well.
+        /// </summary>
+        /// <param name="invalidPartDefnitionsSet">The set of definitions for the invalidated parts.</param>
+        /// <returns>The set of <see cref="ComposedPartDiagnostic"/> with the encountered errors.</returns>
+        public IEnumerable<ComposedPartDiagnostic> CheckForInvalidatedParts(ImmutableHashSet<ComposablePartDefinition> invalidPartDefnitionsSet)
```

Comment:
Taking `ImmutableHashSet` as the concrete type for a parameter ties the hands of your caller. Since the only use of this argument is to call `Contains`, accepting `ICollection<T>` should suffice. That will enable you to avoid creating an immutable collection in your caller, which has allocation benefits.